snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/Snowplow.java

```diff
@@ -0,0 +1,140 @@
+package com.snowplowanalytics.snowplow;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.snowplowanalytics.snowplow.configuration.Configuration;
+import com.snowplowanalytics.snowplow.configuration.NetworkConfiguration;
+import com.snowplowanalytics.snowplow.configuration.TrackerConfiguration;
+import com.snowplowanalytics.snowplow.controller.TrackerController;
+import com.snowplowanalytics.snowplow.internal.tracker.ServiceProvider;
+import com.snowplowanalytics.snowplow.network.HttpMethod;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class Snowplow {
+
+    private static volatile Snowplow singleton;
+
+    // Private properties
+
+    @Nullable
+    private ServiceProvider defaultServiceProvider;
+    @NonNull
+    private final Map<String, ServiceProvider> serviceProviderInstances = new HashMap<>();
+
+    // Singleton
+
+    private Snowplow() {
+        if (singleton != null) {
+            throw new RuntimeException("Use `getInstance` method to create this class.");
```

Comment:
How is this ever going to be fired? You can't call this constructor without calling `getInstance()`?

Reply:
It can be accessed using reflection. But it just may be overkill.